src/managers/AdjacencyManager.ts

```diff
@@ -0,0 +1,76 @@
+import { int, Nullable } from "../definitions/utils";
+import { AdjacencyMap } from "../utilities/AdjacencyMap";
+import tileConfigs from "../configs/tiles.json";
+import { PlaceTile } from "../definitions/placeTile";
+import { SessionDTO } from "../definitions/dto";
+import SessionManager from "./SessionManager";
+import { assert } from "../utilities/utils";
+import { api } from "../utilities/api";
+import TileManager from "./TileManager";
+import { pathRepresentation } from "../definitions/pathRepresentation";
+
+class AdjacencyManager {
+    private adjacencyMap: Nullable<AdjacencyMap>;
+    private pathMap: Map<int, pathRepresentation>;
+    private hasWon: boolean;
+
+    public constructor() {
+        this.adjacencyMap = null;
+        this.pathMap = this.createConnectionsMap();
+        this.hasWon = false;
+    }
+
+    public setHasWon(hasWon: boolean): void {
+        this.hasWon = hasWon;
+    }
+
+    public getHasWon(): boolean {
+        return this.hasWon;
+    }
+    public getPathMap(): Map<int, pathRepresentation> {
+        return this.pathMap;
+    }
+
+    public tile(tile: PlaceTile): void {
```

Comment:
rename to doPlaceTile()? I got confused because i intuitively thought tile is an object at first

Reply:
the manager used to be named place managar but i changed the name and forgot to change the method name sorry